Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -0,0 +1,99 @@
+#if UNITY_EDITOR && UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
+
+using System;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine.InputSystem.Utilities;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    internal static class ProjectWideActionsAsset
+    {
+        internal const string kDefaultAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.inputactions";
+        internal const string kAssetPath = "ProjectSettings/InputManager.asset";
+        internal const string kAssetName = "ProjectWideInputActions";
+
+        static string s_DefaultAssetPath = kDefaultAssetPath;
+        static string s_AssetPath = kAssetPath;
+
+        // For Testing
+        internal static void SetAssetPaths(string defaultAssetPath, string assetPath)
+        {
+            s_DefaultAssetPath = defaultAssetPath;
+            s_AssetPath = assetPath;
+        }
+
+        [InitializeOnLoadMethod]
+        internal static void InstallProjectWideActions()
+        {
+            GetOrCreate();
+        }
+
+        internal static InputActionAsset GetOrCreate()
+        {
+            var objects = AssetDatabase.LoadAllAssetsAtPath(s_AssetPath);
+            if (objects != null)
+            {
+                var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
+                if (inputActionsAsset != null)
+                    return inputActionsAsset;
+            }
+
+            return CreateNewActionAsset();
+        }
+
+        private static InputActionAsset CreateNewActionAsset()
+        {
+            var json = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, s_DefaultAssetPath));
+
+            var asset = ScriptableObject.CreateInstance<InputActionAsset>();
+            asset.LoadFromJson(json);
+            asset.name = kAssetName;
+
+            AssetDatabase.AddObjectToAsset(asset, s_AssetPath);
+
+            // Make sure all the elements in the asset have GUIDs and that they are indeed unique.
```

Comment:
I'd be tempted to split this out into a support function taking the asset as a parameter.

Reply:
In part because I think a 'restore defaults' functionality may use this later
https://jira.unity3d.com/browse/ISX-1564